process/peer/process.go

```diff
@@ -475,14 +475,17 @@ func (vs *validatorStatistics) updateShardDataPeerState(header data.HeaderHandle
 			continue
 		}
 
-		prevShardData, shardInfoErr := process.GetShardHeader(
-			h.PrevHash,
-			vs.dataPool.Headers(),
-			vs.marshalizer,
-			vs.storageService,
-		)
+		prevShardData, shardInfoErr := vs.searchInMap(h.PrevHash, cacheMap)
 		if shardInfoErr != nil {
-			return shardInfoErr
+			prevShardData, shardInfoErr = process.GetShardHeader(
```

Comment:
If not nil, search again and then return?

Reply:
simplified, search only in provided map as you can not start processing if the data is not available in this temporary processing cache.

Reply:
resolved